scripts/examples/typescript/README.md

```diff
@@ -0,0 +1,41 @@
+# A Typescript bindings for XenAPI
+A Typescript bindings for XenAPI, which is inspired by [Python XenAPI](https://xapi-project.github.io/xen-api/usage.html).
+
+## Usage
+Install lib from npm repository
+```
+$ npm install xen-api-ts
+```
+
+The usage of Typescript XenAPI is almost identical to the Python XenAPI, except it's asynchronous and requires async/await.
+```
+import { xapi_client } from "xen-api-ts";
+
+async function main() {
+  const session = xapi_client(process.env.HOST_URL);
+  try {
+    await session.login_with_password(process.env.USERNAME, process.env.PASSWORD);
+    const hosts = await session.xenapi.host.get_all();
+    console.log(hosts); // Do something with the retrieved hosts
+  } finally {
+    await session.xenapi.session.logout();
+  }
+}
+
+main();
+```
+
+For more example usage, please find in tests folder.
+```
+$ git clone git@github.com:acefei/xen-api-ts.git
```

Comment:
To make things simpler for the users, we're trying to minimise the number of places where people can find xenserver/xen-api documentation and help. You may want to consider moving your examples (importing the whole git history if you want) to a new folder `typescript` within this repo https://github.com/xenserver/xenserver-samples which is also linked from the official XenServer developer documentation. Let me know and I'll add you to the maintainers list.